src/components/selector_menu/index.js

```diff
@@ -128,23 +126,30 @@ var SelectorMenu = React.createClass({
       return;
     }
 
-    var visible = _.map(fuzzy.filter(filterBy, this.state.visible), function(result) {
-      return result.string;
-    });
+    var visible = _.pluck(fuzzy.filter(filterBy, this.getOptionNames()), 'string');
 
     this.setState({
       visible: visible
     });
   },
 
+  componentWillReceiveProps: function(nextProps) {
+    if (nextProps.optionsList.length !== this.props.optionsList.length) {
```

Comment:
Is there a case here where they have the same length, but have different contents?